lib/storage/src/content_manager/consensus/persistent.rs

```diff
@@ -235,10 +258,14 @@ impl Persistent {
     ///
     /// `first_peer` - if this is a first peer in a new deployment (e.g. it does not bootstrap from anyone)
     /// It is `None` if distributed deployment is disabled
-    fn init(path: PathBuf, first_peer: bool) -> Result<Self, StorageError> {
+    fn init(
+        path: PathBuf,
+        first_peer: bool,
+        this_peer_id: Option<PeerId>,
+    ) -> Result<Self, StorageError> {
         // Do not generate too big peer ID, to avoid problems with serialization
         // (especially in json format)
-        let this_peer_id = rand::random::<PeerId>() % (1 << 53);
+        let this_peer_id = this_peer_id.unwrap_or_else(|| rand::random::<PeerId>() % (1 << 53));
```

Comment:
Unrelated but, do we do `% (1 << 53)` here to not use numbers higher than JavaScript supports?

Reply:
exactly because of JavaScript

Reply:
Ideally we should have used a uuid, but migration overhead is too big now